### PR TITLE
[core IR] Add lift_fresh, split.Tensor, and unbind decompositions to core ATen decomp table

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110102

## Context

Add existing decomps for `lift_fresh`, `split.Tensor`, and `unbind` to the core ATen decomposition table. Do not use them in inductor, since Inductor currently lowers these directly.

One note though is that `lift_fresh`'s decomposition has a note saying it's not correct under autograd. However, my understanding is that these decompositions are registered to the `"post_autograd"` decomposition table, meaning autograd wouldn't be a factor. Would like some confirmation that this premise is correct.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_decomp/__init__.py [hunk 1]
```diff
@@ -328,6 +328,7 @@ def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
             aten.leaky_relu_backward,
             aten.lerp,
             aten.lerp_,
+            aten.lift_fresh,
```
Comment (lezcano) [inline torch/_decomp/__init__.py:331]: The lift_fresh decomposition is just a noop, and it has a note saying that it's not correct under autograd. Do we want to add it as a core decomp?

Comment (SS-JIA) [reply torch/_decomp/__init__.py:331]: My understanding is that the core decompositions are registered to the `"post_autograd"` decomposition table (see `torch/_decomp/__init__.py`), presumably that means that these decompositions should be used when autograd is not a factor.

This is a good point though, would like to get additional confirmation that what I said above makes sense.

cc: @SherlockNoMad @bdhirsh @jansel

**Review by jansel — APPROVED** [review]

File: torch/_decomp/__init__.py (hunk 1 above)
Comment (bdhirsh) [reply torch/_decomp/__init__.py:331]: hmm @ezyang - today when we see a tensor constant during tracing, that turns into a `constant_lifted = aten.lift_fresh.default(self._tensor_constant0)` in the graph. Do you think `lift_fresh` should be a first-class concept for backends?

Comment (bdhirsh) [reply torch/_decomp/__init__.py:331]: we guarantee that a graph of core ATen IR is always functional (right @SherlockNoMad?). Then it sounds reasonable for core aten to no-op lift_fresh.

File: benchmarks/dynamo/ci_expected_accuracy/inductor_torchbench_training.csv [hunk 2]
```diff
@@ -50,5 +50,5 @@ timm_vision_transformer_large,pass_due_to_skip,0
 timm_vovnet,pass,8
 tts_angular,pass,10
 vgg16,pass,8
-vision_maskrcnn,fail_accuracy,42
+vision_maskrcnn,fail_accuracy,39
```
Comment (bdhirsh) [inline benchmarks/dynamo/ci_expected_accuracy/inductor_torchbench_training.csv:53]: any idea why this PR affected # of graph breaks in vision_maskrcnn? 🤔

Comment (SS-JIA) [reply benchmarks/dynamo/ci_expected_accuracy/inductor_torchbench_training.csv:53]: So the actual cause of this was https://github.com/pytorch/pytorch/pull/110160.

Not sure why it was showing a failure for this PR though...

**Pull request closed without merging**